src/IO/ITKImageInputFileFormat.cxx

```diff
@@ -81,46 +109,124 @@ ITKImageInputFileFormat::can_read(const FileSignature& /*signature*/,
 {
   return
-    unique_ptr<DiscretisedDensity<3,float> >
-    (read_file_itk< FinalImageType >(filename));
+    unique_ptr<STIRImageType>
+    (read_file_itk< STIRImageType >(filename));
+}
+
+// Actual conversion function
+STIRImageSingle*
+convert_ITK_to_STIR(const ITKImageSingle::Pointer itk_image_orig)
+{
+    ITKImageSingle::Pointer itk_image;
+    CartesianCoordinate3D<float> voxel_size;
+    BasicCoordinate<3,int> min_indices, max_indices;
+    CartesianCoordinate3D<float> origin;
+
+    // orientate the ITK image
+    orient_ITK_image<ITKImageSingle>(
+                itk_image, voxel_size, min_indices, max_indices, origin, itk_image_orig);
+
+    // create STIR image
+    VoxelsOnCartesianGrid<float>* image_ptr =
+        new VoxelsOnCartesianGrid<float>
+            (IndexRange<3>(min_indices, max_indices),
+            origin,
+            voxel_size);
+
+    // copy data
+    VoxelsOnCartesianGrid<float>::full_iterator stir_iter = image_ptr->begin_all();
+    typedef itk::ImageRegionConstIterator< ITKImageSingle > IteratorType;
+    IteratorType it (itk_image, itk_image->GetLargestPossibleRegion() );
+    for ( it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter  ) {
+        *stir_iter = static_cast<float>(it.Get());
+    }
+
+    return image_ptr;
 }
 
+// Actual conversion function
+STIRImageMulti*
+convert_ITK_to_STIR(const ITKImageMulti::Pointer itk_image_orig)
+{
+    ITKImageMulti::Pointer itk_image;
+    CartesianCoordinate3D<float> voxel_size;
+    BasicCoordinate<3,int> min_indices, max_indices;
+    CartesianCoordinate3D<float> origin;
+
+    // orientate the ITK image
+    orient_ITK_image<ITKImageMulti>(
+                itk_image, voxel_size, min_indices, max_indices, origin, itk_image_orig);
+
+    // create STIR image
+    STIRImageMulti* image_ptr =
+        new STIRImageMulti
+            (IndexRange<3>(min_indices, max_indices),
+             origin,
+             voxel_size);
+
+    // copy data
+    STIRImageMulti::full_iterator stir_iter = image_ptr->begin_all();
+    typedef itk::ImageRegionConstIterator< ITKImageMulti > IteratorType;
+    IteratorType it (itk_image, itk_image->GetLargestPossibleRegion() );
+    for ( it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter) {
+        itk::Point<double,3U> itk_coord;
+        itk_coord[0] = -it.Get()[0]; // Seems we need an extra flip in the x-axis. don't know why...
+        itk_coord[1] = -it.Get()[1]; // Seems we need an extra flip in the y-axis. don't know why...
+        itk_coord[2] =  it.Get()[2];
+        *stir_iter =
+            ITK_coordinates_to_STIR(
+                itk_coord,
+                voxel_size,
+                min_indices,
+                true);
+    }
+
+    return image_ptr;
+}
 
 //To read any file format via ITK
-template<typename TImageType> VoxelsOnCartesianGrid<float>*
+template<>
+STIRImageSingle*
 read_file_itk(std::string filename)
 {
+#ifndef NDEBUG
+  info(boost::format("Opening a single image: %1%") % filename);
```

Comment:
I guess you used this as debugging. the message is rather cryptic and won't work anymore if it's back to 1 template. Delete, or generalise to something like `Using ITK to read ...` (but then probably more useful to have a different message for DICOM and the normal stuff).